server/src/test/java/io/crate/integrationtests/CopyIntegrationTest.java

```diff
@@ -1133,4 +1133,35 @@ public void test_copy_preserves_the_implied_sub_column_order() throws IOExceptio
                 ")"
             );
     }
+
+    @Test
+    public void test_generated_non_deterministic_value_is_consistent_on_primary_and_replica() throws Exception {
```

Comment:
This is off the track(caught my attention and got carried away...) but `shared=true` causes one of `FileReadingIterator.advanceToNextFileInput` to throw `IOException("Cannot find any URI matching: " + uri.toString())` eventually causing `RawIndexer.index` to throw NPE at line 2 - `XContentHelper.convertToMap(JsonXContent.JSON_XCONTENT, raw, true);` Then the NPE is caught by https://github.com/crate/crate/blob/8ad0c3d1cfadd6c2b19dee1d5f2313fe2afc7a11/server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java#L225

Reply:
Sorry should be unrelated to this PR

Reply:
AFAIK there no harm.

It because I'm using `*.json` in the file name (like we do in many other tests) and in falls into 
```
else if (currentInput.isGlobbed()) {`
 ....
 throw new IOException("Cannot find any URI matching: " + uri.toString());
}
```
to my understanding it's a way to stop the * lookup in directory (if read up all files)